database/migrations/default/1765230748998_crash_address_display_trigger/up.sql

```diff
@@ -0,0 +1,776 @@
+    END IF;
+    
+    -- Add street name if provided and not empty
+    IF street_name IS NOT NULL AND street_name != '' THEN
+        formatted_address := formatted_address || street_name || ' ';
+    END IF;
+    
+    -- Add street suffix if provided and not empty
+    IF street_sfx IS NOT NULL AND street_sfx != '' THEN
+        formatted_address := formatted_address || street_sfx || ' ';
+    END IF;
+    
+    -- Remove trailing space
+    RETURN TRIM(formatted_address);
+END;
+$$ LANGUAGE plpgsql;
+
+
+
+-- Function to remove duplicate parts of address
+CREATE OR REPLACE FUNCTION deduplicate_address_parts(address TEXT) RETURNS TEXT AS $$
+DECLARE
+    parts TEXT[];
+    deduped_parts TEXT[] := '{}';
+    i INT;
+    current_part TEXT;
+    prev_part TEXT;
+    prev_upper TEXT;
+BEGIN
+    -- Split address by spaces
+    parts := string_to_array(address, ' ');
+    
+    FOR i IN 1..array_length(parts, 1) LOOP
+        current_part := parts[i];
+        
+    IF i = 1 THEN
+            -- First word of address always included
+            deduped_parts := array_append(deduped_parts, current_part);
+        ELSE
+            prev_part := parts[i-1];
+            
+            -- Only add if not a duplicate of previous (case-insensitive)
+            IF UPPER(current_part) != UPPER(prev_part) THEN
+                deduped_parts := array_append(deduped_parts, current_part);
+            END IF;
+        END IF;
+    END LOOP;
+    
+    -- Reconstruct address
+    RETURN array_to_string(deduped_parts, ' ');
+END;
+$$ LANGUAGE plpgsql;
+
+
+-- Main function to generate complete crash address
+CREATE OR REPLACE FUNCTION generate_crash_address(
+    at_intrsct_fl BOOLEAN,
+    -- Primary address parts
+    rpt_block_num TEXT,
+    rpt_street_pfx TEXT,
+    rpt_street_name TEXT,
+    rpt_street_sfx TEXT,
+    rpt_hwy_num TEXT,
+    rpt_hwy_sfx TEXT,
+    rpt_rdwy_sys_id INTEGER,
+    rpt_road_part_id INTEGER,
+    -- Secondary address parts
+    rpt_sec_block_num TEXT,
+    rpt_sec_street_pfx TEXT,
+    rpt_sec_street_name TEXT,
+    rpt_sec_street_sfx TEXT,
+    rpt_sec_hwy_num TEXT,
+    rpt_sec_hwy_sfx TEXT,
+    rpt_sec_rdwy_sys_id INTEGER,
+    rpt_sec_road_part_id INTEGER
+) RETURNS TEXT AS $$
+DECLARE
+    primary_address TEXT := '';
+    secondary_address TEXT := '';
+    rwy_sys_primary_label TEXT;
+    rwy_sys_secondary_label TEXT;
+    road_part_primary_label TEXT;
+    road_part_secondary_label TEXT;
+    formatted_address TEXT := '';
+BEGIN
+    -- Get lookup values
+    SELECT label INTO rwy_sys_primary_label 
+    FROM lookups.rwy_sys WHERE id = rpt_rdwy_sys_id;
+    
+    SELECT label INTO rwy_sys_secondary_label 
+    FROM lookups.rwy_sys WHERE id = rpt_sec_rdwy_sys_id;
+    
+    SELECT label INTO road_part_primary_label 
+    FROM lookups.road_part WHERE id = rpt_road_part_id;
+    
+    SELECT label INTO road_part_secondary_label 
+    FROM lookups.road_part WHERE id = rpt_sec_road_part_id;
```

Comment:
Maybe the performance hit is negligible, but I think it'd be fine to code all these functions on the ID values (`rpt_rdwy_sys_id`, `rpt_sec_rdwy_sys_id`, etc) instead of joining up the labels 🤔